extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/dev/DevServicesOpenTelemetryProcessor.java

```diff
@@ -20,43 +21,37 @@
 public class DevServicesOpenTelemetryProcessor {
 
     @BuildStep
-    void devServicesDatasources(
-            Optional<DevServicesDatasourceResultBuildItem> devServicesDatasources,
+    void devServicesDatasources(List<JdbcDataSourceBuildItem> jdbcDataSources,
             BuildProducer<DevServicesAdditionalConfigBuildItem> devServicesAdditionalConfig) {
-
-        if (devServicesDatasources.isPresent()) {
-            Map<String, String> properties = new HashMap<>();
-            DevServicesDatasourceResultBuildItem.DbResult defaultDatasource = devServicesDatasources.get()
-                    .getDefaultDatasource();
-            if (defaultDatasource != null) {
-                properties.putAll(defaultDatasource.getConfigProperties());
-            }
-
-            Map<String, DevServicesDatasourceResultBuildItem.DbResult> namedDatasources = devServicesDatasources.get()
-                    .getNamedDatasources();
-            if (namedDatasources != null) {
-                for (DevServicesDatasourceResultBuildItem.DbResult dbResult : namedDatasources.values()) {
-                    if (dbResult != null) {
-                        properties.putAll(dbResult.getConfigProperties());
+        for (JdbcDataSourceBuildItem dataSource : jdbcDataSources) {
+            // if the datasource is explicitly configured to use the OTel driver...
+            if (dataSourceUsesOTelJdbcDriver(dataSource.getName())) {
+                List<String> urlPropertyKeys = DataSourceUtil.dataSourcePropertyKeys(dataSource.getName(), "jdbc.url");
+                devServicesAdditionalConfig.produce(new DevServicesAdditionalConfigBuildItem(devServicesConfig -> {
+                    Map<String, String> overrides = new HashMap<>();
+                    for (String key : urlPropertyKeys) {
+                        String devServicesUrl = devServicesConfig.get(key);
+                        // ... and if the same datasource uses dev services...
+                        if (devServicesUrl != null) {
+                            // ... then we rewrite the jdbc url to add the otel prefix.
+                            overrides.put(key, devServicesUrl.replaceFirst("jdbc:", "jdbc:otel:"));
```

Comment:
Could we test if the url already starts with `jdbc:otel:` before doing this change. I noticed it before and somehow forgot about it. Looks like a good opportunity to fix it.

(The users could set the Otel URL themselves)

Reply:
This is the URL set automatically by dev services. The users don't have control over that, do they?

Reply:
Ah, yes, you're right. Sorry about the noise.

Reply:
I just checked, and the Dev Services JDBC prefixes are indeed hardcoded, e.g.:

https://github.com/quarkusio/quarkus/blob/c7b90ee0d404a87422dcc44b5db5acdbf9efd67e/extensions/devservices/h2/src/main/java/io/quarkus/devservices/h2/deployment/H2DevServicesProcessor.java#L60

https://github.com/quarkusio/quarkus/blob/c7b90ee0d404a87422dcc44b5db5acdbf9efd67e/extensions/devservices/mariadb/src/main/java/io/quarkus/devservices/mariadb/deployment/MariaDBDevServicesProcessor.java#L112

Etc.

So... I don't think it's possible that we get `jdbc:otel` in this URL with the current code. And I don't think it would make sense to allow users to provide the JDBC URL themselves when using dev services, since by definition dev services are started by Quarkus and users don't know on what port they run.